src/main.cpp

```diff
@@ -53,7 +54,104 @@ void main_loop(const char *filename, unsigned asset) {
+        fprintf(stdout, "Research Item:\t\t%d\n",
+            state->_players[i].researchItem);
+
+        fprintf(stdout, "--- Racepicks ---\n");
+        fprintf(stdout, "Government:\t\t%d\tPopulation:\t\t%d\tFarming:\t\t%d\tScience:\t\t%d\n",
+            state->_players[i].racePicks.government, state->_players[i].racePicks.population,
+            state->_players[i].racePicks.farming, state->_players[i].racePicks.science);
+        fprintf(stdout, "Money:\t\t\t%d\tShip defense:\t\t%d\tShip attack:\t\t%d\tGround combat:\t\t%d\n",
+            state->_players[i].racePicks.money, state->_players[i].racePicks.shipDefense,
+            state->_players[i].racePicks.shipAttack, state->_players[i].racePicks.groundCombat);
+        fprintf(stdout, "Spying:\t\t\t%d\tLow G:\t\t\t%d\tHigh G:\t\t\t%d\tAquatic:\t\t%d\n",
+            state->_players[i].racePicks.spying, state->_players[i].racePicks.lowG,
+            state->_players[i].racePicks.highG, state->_players[i].racePicks.aquatic);
+        fprintf(stdout, "Subterranian:\t\t%d\tLarge homeworld:\t%d\tRich/Poor homeworld:\t%d\tArtifacts homeworld:\t%d\n",
+            state->_players[i].racePicks.subterranean, state->_players[i].racePicks.largeHomeworld,
+            state->_players[i].racePicks.richHomeworld, state->_players[i].racePicks.artifactsHomeworld);
+        fprintf(stdout, "Cybernetic:\t\t%d\tLithovore:\t\t%d\tRepulsive:\t\t%d\tCharismatic:\t\t%d\n",
+            state->_players[i].racePicks.cybernetic, state->_players[i].racePicks.lithovore,
+            state->_players[i].racePicks.repulsive, state->_players[i].racePicks.charismatic);
+        fprintf(stdout, "Uncreative:\t\t%d\tCreative:\t\t%d\tTolerant:\t\t%d\tFantastic traders:\t%d\n",
+            state->_players[i].racePicks.uncreative, state->_players[i].racePicks.creative,
+            state->_players[i].racePicks.tolerant, state->_players[i].racePicks.fantasticTraders);
+        fprintf(stdout, "Telepathic:\t\t%d\tLucky:\t\t\t%d\tOmniscience:\t\t%d\tStealthy ships:\t\t%d\n",
+            state->_players[i].racePicks.telepathic, state->_players[i].racePicks.lucky,
+            state->_players[i].racePicks.omniscience, state->_players[i].racePicks.stealthyShips);
+        fprintf(stdout, "Transdimensional:\t%d\tWarlord:\t\t%d\n",
+            state->_players[i].racePicks.transDimensional, state->_players[i].racePicks.warlord);
+    }
+
+    fprintf(stdout, "Number of stars: %d\n", state->_starCount);
+    for (auto star : state->_stars) {
+        fprintf(stdout, "\nName:\t%s\n", star.name);
+        fprintf(stdout, "Class:\t\t%d\t\tSize:\t\t%d\n", star.spectralClass, star.size);
+        fprintf(stdout, "Position:\t%d,%d\tPrimary owner:\t%d\n", star.x, star.y, star.owner);
+        fprintf(stdout, "Special:\t%d\t\tWormhole:\t%d\n", star.special, star.wormhole);
+    }
+}
+
 int main(int argc, char **argv) {
+    Savegame saveGame("/home/ddonisa/Games/master-of-orion-2/data/SAVE1.GAM");
```

Comment:
It'd be easier to pass the path through `argv[1]`, the asset loading code below is just for testing anyway so feel free to rewrite it as much as you need. And you don't need to commit purely experimental code.